modules/ilab/main.tf

```diff
@@ -0,0 +1,78 @@
+#
+# Developer tips:
+#   - Below code should be replaced with the code for the root level module
+#
+
+##############################################################################
+# Data - SSH Private Key
+##############################################################################
+
+data "ibm_sm_arbitrary_secret" "sshkey" {
+  instance_id   = "${var.sm_instance_id}"
+  region        = "${var.region}"
+  secret_id     = "${var.sm_ssh_private_key_id}"
+}
+
+##############################################################################
+# Write the SSH key to a temporary file
+# Make the file only readable by the owner
+##############################################################################
+
+resource "local_file" "ssh_key" {
```

Comment:
Consider using local_sensitive_file for this key: https://registry.terraform.io/providers/hashicorp/local/latest/docs/resources/sensitive_file